curvefs/src/client/curve_fuse_op.cpp

```diff
@@ -199,15 +216,22 @@ void FuseOpOpen(fuse_req_t req, fuse_ino_t ino, struct fuse_file_info *fi) {
 void FuseOpRead(fuse_req_t req, fuse_ino_t ino, size_t size, off_t off,
                 struct fuse_file_info *fi) {
     std::unique_ptr<char[]> buffer(new char[size]);
-    memset(buffer.get(), 0, size);
     size_t rSize = 0;
     CURVEFS_ERROR ret = g_ClientInstance->FuseOpRead(req, ino, size, off, fi,
                                                      buffer.get(), &rSize);
     if (ret != CURVEFS_ERROR::OK) {
         FuseReplyErrByErrCode(req, ret);
         return;
     }
-    fuse_reply_buf(req, buffer.get(), rSize);
+
+    struct fuse_bufvec bufvec;
+    bufvec.count = 1;
+    bufvec.off = 0;
+    bufvec.idx = 0;
+    bufvec.buf[0].size = rSize;
+    bufvec.buf[0].mem = buffer.get();
```

Comment:
Here data in buffer is in user space, should we use bufvec instead of std::unique_ptr<char[]> buffer(new char[size]);?

Reply:
no, they are two different things. bufvec is a vector of fuse_buf, and fuse_buf has a pointer `mem` that points the data that we want to send to kernel space. so, we have to allocate a buffer to store read data and make `mem` points to it.